api/src/main/java/graphql/nadel/schema/NadelDirectives.java

```diff
@@ -171,48 +287,91 @@ private static Description createDescription(String s) {
 
+    private static UnderlyingServiceHydration buildHydrationParameters(GraphQLDirective directive, List<RemoteArgumentDefinition> arguments) {
+        String service = getDirectiveValue(directive, "service", String.class);
+        String field = getDirectiveValue(directive, "field", String.class);
+        String objectIdentifier = getDirectiveValue(directive, "identifiedBy", String.class);
+        Boolean objectIndexed = getDirectiveValue(directive, "indexed", Boolean.class, false);
+        // Note: this is not properly implemented yet, so the value does not matter
+        boolean batched = false; // getDirectiveValue(directive, "batched", Boolean.class, false);
+        if (objectIndexed) {
+            objectIdentifier = null; // we cant have both but it has a default
+        }
+        int batchSize = getDirectiveValue(directive, "batchSize", Integer.class, 200);
+        int timeout = getDirectiveValue(directive, "timeout", Integer.class, -1);
+
+        List<String> fieldNames = dottedString(field);
+        assertTrue(fieldNames.size() >= 1);
+        String topLevelFieldName = fieldNames.get(0);
+        String syntheticField = null;
+        if (fieldNames.size() > 1) {
+            syntheticField = fieldNames.get(0);
+            topLevelFieldName = fieldNames.get(1);
+        }
+
+        // nominally this should be some other data class that's not an AST element
+        // but history is what it is, and it's an AST element that's' really a data class
+        return new UnderlyingServiceHydration(
+                emptySrc(),
+                emptyList(),
+                service,
+                topLevelFieldName,
+                syntheticField,
+                arguments,
+                objectIdentifier,
+                objectIndexed,
+                batched,
+                batchSize,
+                timeout,
+                emptyMap()
+        );
+    }
+
+    private static UnderlyingServiceHydration createTemplatedUnderlyingServiceHydration(GraphQLDirective hydratedFromDirective, GraphQLSchema overallSchema) {
+
+        GraphQLArgument template = hydratedFromDirective.getArgument("template");
+        String enumTargetName = resolveArgumentValue(template);
+        GraphQLEnumType templateEnumType = overallSchema.getTypeAs("NadelHydrationTemplate");
+        assertNotNull(templateEnumType, () -> "There MUST be a enum called NadelHydrationTemplate");
+
+        GraphQLEnumValueDefinition enumValue = templateEnumType.getValue(enumTargetName);
+        assertNotNull(enumValue, () -> String.format("There MUST be a enum value in NadelHydrationTemplate called '%s'", enumTargetName));
+
+        GraphQLDirective templateDirective = enumValue.getDirective(HYDRATED_TEMPLATE_DIRECTIVE_DEFINITION.getName());
+        assertNotNull(templateDirective, () -> String.format("The enum value '%s' in NadelHydrationTemplate must have a directive called '%s'", enumTargetName, HYDRATED_TEMPLATE_DIRECTIVE_DEFINITION.getName()));
```

Comment:
Just an observation, this will just error out the validation and return a stacktrace.

It'll have to do for now, but I think later down the line we should have something like

`NadelDirectives` just parses syntax to POJO
Validation works on the POJO.
Blueprint takes the POJO and creates instructions with things resolved and wired up, with the assumption that validation has passed and it's all good.

Reply:
Yes for latter it could be just the capture POJO details.  This is migratory